proposals/1113_ldap-ad-web-ui/README.md

```diff
@@ -0,0 +1,136 @@
+IDP is an LDAP or Active Directory IDP, the user will be redirected to the new login page.
+When a client other than the Pinniped CLI makes an authorization endpoint request with
+custom Username/Password headers, they should be rejected.
+
+The pinniped cli should default to using the cli-based password flow, but when a tty is unavailable or when
+a CLI flag (`--ldap-browser-login`? `--force-browser-login`?) is specified, it will open a browser to log in
+instead of prompting for username and password. Some users (for example, IDE plugins for kubernetes)
+may wish to authenticate using the pinniped cli but without access to a terminal.
+
+#### Upgrades
+
+This change is backwards compatible. Users would see no changes unless they decided to register
+a new client or change the pinniped cli flags.
+
+However if they do choose to register a new client they may need to update the following:
+- FederationDomains today may be using private certificate authorities. These are trusted
+  for our use case but a browser will flag them as unsafe. Admins will have to transition to letsencrypt
+  or another public Certificate Authority to prevent making end users click past messages about the certificate
+  being untrusted.
+- The name of the idp custom resource is currently not published to users logging in with Pinniped.
+  We plan on exposing this to indicate to users which idp they are logging in to.
+  Admins may need to update this to something more user-friendly.
+
+#### Tests
+
+Chromedriver browser based integration tests will be needed to ensure that a user can log in from a web-based app
+by entering their ldap credentials into the web page.
+
+#### New Dependencies
+This should be kept to a very simple HTML page with minimal, clean CSS styling.
+Javascript should be avoided.
+
+#### Observability Considerations
+* Logging login attempts at higher log levels.
+
+#### Security Considerations
+* Preventing LDAP injection attacks: this should be done server-side using our existing
+  string escaping.
+* CSRF protection via a CSRF cookie: this should be similar to the way it is done for the
+  OIDCIdentityProvider today
+* The new UI page must be HTTPS.
+
+#### Documentation Considerations
+This new feature will require documentation to explain how to configure it and to publicise that it is available.
+This should include:
+* A blog post describing the feature
+* Website documentation in the form of a how-to guide
+
+### Other Approaches Considered
+Today, users can configure Dex if they want a web-based LDAP login.
+This introduces complexity because they have to install, configure and
+maintain both Pinniped and Dex in order to use this feature. It also means
+that users do not benefit from the opinionated `ActiveDirectoryIdentityProvider`
+config because Dex does not have an equivalent.
+
+## Open Questions
+* Is the ldap login page hosted per FederationDomain?
```

Comment:
It does not really matter today because you can only have one IDP.

Reply:
How does the CLI handle this today? The CLI OAuth client isn't associated with a federation domain.  What about the APIs that orchestrate the upstream login, are they associated with a federation domain?

Reply:
The CLI does not really know anything about a federation domain - it just looks at the issuer URL to perform discovery to find the generic OIDC endpoints and the pinniped specific endpoints.  In that sense, all end-user facing supervisor APIs are associated with a federation domain because there simply is not any other place to put them.

Reply:
I removed this as an open question because I agree with Mo that it does not matter much today as we have one IDP, and the question felt too broad.
At the moment I think there should be one login page hosted at `issuer-url/some/path`, which in practice means it is associated with a FederationDomain as that's where the issuer is defined. Does that make sense to folks or do we still need to talk through this more?

Reply:
I think that makes sense. By putting it on the FederationDomain's endpoints, then the TLS serving certs will be handled correctly by our existing configuration options and implementation code. Note that TLS serving certs are already per-FederationDomain.